### DISCUSSION TITLE
What is the recommended way to install on RHEL 8?

### DISCUSSION THREAD

**Discussion by ghefler** [MISP/MISP#9857] [Q&A]
Hi, currently trying to install MISP server on RHEL (version 8.9) with some issues 🥲

Problems identified with install [script](https://misp.github.io/MISP/INSTALL.rhel8): 

1. module remi not being available for RHEL 8.9. The following msg gets printed: `missing groups or modules: php:remi-7.4` even though PHP v7.4 is installed. Not familiar with PHP ecosystem of how different versions of it are managed/supported but PHP is definitely installed.
2. Python 3.6 when trying to install `misp-stix` - `/var/www/MISP/venv` : unable to find a valid version of `Poetry_core>=1.2`
3. SElinux permission errors when starting `httpd.service`
4. Bunch of errors related to `PHP Warning:  require_once(/var/www/MISP/app/Vendor/autoload.php): failed to open stream: No such file or directory in /var/www/MISP/app/Config/core.php on line 298` 

There is/was issue with SELinux blocking `/var/www/MISP/app/Console/worker/start.sh` execution. Error logs suggested solution was to generate a policy with `ausearch -c '(start.sh)' --raw | audit2allow -M my-startsh` and `semodule -X 300 -i my-startsh.pp`  when starting `httpd.service`. However, there seems to be an issue with `start.sh` as it outputs `file not found` or `permission denied`.  Running the script as `apache` user with `su -u apache bash /var/www/MISP/app/Console/worker/start.sh` seem to work be we are getting the following error: 

`PHP Fatal error:  require_once(): Failed opening required '/var/www/MISP/app/Vendor/autoload.php' (include_path='/var/www/MISP/app/Lib/cakephp/lib:.:/usr/share/pear:/usr/share/php') in /var/www/MISP/app/Config/core.php on line 298`

I assume either PHP installation messed up or this still a permissions errror.

Anyone with similar issues?

**Comment by adulau** [comment]
Maybe you want to have a look at https://github.com/misp/misp-rpm ?

**Reply by ghefler** [reply]
Thanks, I'll have a look. 

I'm guessing the requirements for MISP server to work are:

1. Apache
2. PHP 7.4
3. Python 3.6  - does it work with newer versions of Python?
4. MariaDB

Tested last night on a RedHat UBI8 container and managed to make progress installing project related dependencies using the default repos. However, still have some teething issues related to UBI.
